### PR TITLE
aws/data_stream/route53_public_logs: Handle non-%HOSTNAME entries

### PR DESCRIPTION
**Pull request by cafuego**
## Proposed commit message

Change the HOSTNAME pattern to DATA in the grok snippet to match route53 public logs, so that DNS records that contain characters not included in `[0-9A-Za-z][0-9A-Za-z-]` can still be processed.

This notably includes entries such as _dmarc, _spf and various types of domain ownership verification records.

Those are a non-exhaustive list that are being dropped from my logs, but note that the DNS spec doesn't prevent other characters being present. The spec does not inform what a client might *query* for, which is what we're dealing with, not a zone file.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## How to test this PR locally

Use the provided patterns in `packages/aws/data_stream/route53_public_logs/_dev/test/pipeline/test-route53.log`

### CODE REVIEW

**Comment by cafuego** [comment]
Yup, still relevant and does not contain a dependency on xz 🤪

**Comment by ishleenk17** [comment]
@cafuego : cAn we please resolve the merge conflicts

File: packages/aws/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.15.3"
```
Comment (efd6) [inline packages/aws/changelog.yml:2]:
```suggestion
- version: "2.16.0"
```
(bump minor for enhancements)

File: packages/aws/data_stream/route53_public_logs/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ processors:
   - grok:
       field: event.original
       patterns:
-        - '%{BASE10NUM} %{TIMESTAMP_ISO8601:_tmp.timestamp} %{DATA:aws.route53.hosted_zone_id} %{HOSTNAME:_tmp.question} %{WORD:dns.question.type} %{WORD:dns.response_code} %{WORD:network.transport} %{EDGE_LOCATION:aws.route53.edge_location} %{IP:source.address} (%{SUBNET:aws.route53.edns_client_subnet}|-)'
+        - '%{BASE10NUM} %{TIMESTAMP_ISO8601:_tmp.timestamp} %{DATA:aws.route53.hosted_zone_id} %{DATA:_tmp.question} %{WORD:dns.question.type} %{WORD:dns.response_code} %{WORD:network.transport} %{EDGE_LOCATION:aws.route53.edge_location} %{IP:source.address} (%{SUBNET:aws.route53.edns_client_subnet}|-)'
       pattern_definitions:
```
Comment (efd6) [inline packages/aws/data_stream/route53_public_logs/elasticsearch/ingest_pipeline/default.yml:35]: The alternative would be to provided an augmented hostname pattern that includes all the codepoints that occur in DNS names. This only requires that we add `_` to the possible codepoints.

```suggestion
        - '%{BASE10NUM} %{TIMESTAMP_ISO8601:_tmp.timestamp} %{DATA:aws.route53.hosted_zone_id} %{DNS_QUESTION:_tmp.question} %{WORD:dns.question.type} %{WORD:dns.response_code} %{WORD:network.transport} %{EDGE_LOCATION:aws.route53.edge_location} %{IP:source.address} (%{SUBNET:aws.route53.edns_client_subnet}|-)'
      pattern_definitions:
        DNS_QUESTION: \b(?:[_0-9A-Za-z][_0-9A-Za-z-]{0,62})(?:\.(?:[_0-9A-Za-z][_0-9A-Za-z-]{0,62}))*(\.?|\b)
```

Comment (cafuego) [reply packages/aws/data_stream/route53_public_logs/elasticsearch/ingest_pipeline/default.yml:35]: At the very least you also need to add an escaped backslash, or you'll still miss any encoded UTF-8 data.

DNS entries can also contain any binary rubbish you'd care to put in (although Route53 may not allow you to add that) so from a log analysis perspective it's important to grab those logs and allow them to be analysed.

Comment (efd6) [reply packages/aws/data_stream/route53_public_logs/elasticsearch/ingest_pipeline/default.yml:35]: > DNS entries can also contain any binary rubbish you'd care to put in

That's horrifying, but point well made.

File: packages/aws/data_stream/route53_public_logs/_dev/test/pipeline/test-route53.log-expected.json [hunk 3]
```diff
@@ -440,6 +440,81 @@
...
+            },
+            "ecs": {
+                "version": "8.0.0"
+            },
+            "event": {
+                "category": [
+                    "network"
+                ],
+                "kind": "event",
+                "original": "1.0 2017-12-13T08:17:05.744Z Z123412341234 _spf.example.com SPF NOERROR UDP DEN50-C1 2607:f8b0:4001:c07::109 -",
```
Comment (efd6) [inline packages/aws/data_stream/route53_public_logs/_dev/test/pipeline/test-route53.log-expected.json:473]:
```suggestion
                "version": "8.11.0"
            },
            "event": {
                "category": [
                    "network"
                ],
                "kind": "event",
                "original": "1.0 2017-12-13T08:17:05.744Z Z123412341234 _spf.example.com SPF NOERROR UDP DEN50-C1 89.160.20.112 -",
```
from CI:
```diff
test case failed: Expected results are different from actual ones: --- want
+++ got
@@ -463,14 +463,14 @@
                 "response_code": "NOERROR"
             },
             "ecs": {
-                "version": "8.0.0"
-            },
-            "event": {
-                "category": [
-                    "network"
-                ],
-                "kind": "event",
-                "original": "1.0 2017-12-13T08:17:05.744Z Z123412341234 _spf.example.com SPF NOERROR UDP DEN50-C1 2607:f8b0:4001:c07::109 -",
+                "version": "8.11.0"
+            },
+            "event": {
+                "category": [
+                    "network"
+                ],
+                "kind": "event",
+                "original": "1.0 2017-12-13T08:17:05.744Z Z123412341234 _spf.example.com SPF NOERROR UDP DEN50-C1 89.160.20.112 -",
                 "outcome": "success",
                 "type": [
                     "protocol"
```

**Comment by cafuego** [comment]
Aww, sad trombone. I'll update the test.

**Comment by cafuego** [comment]
@efd6 thanks mate!

**Pull request merged**